SimpleLogin/SimpleLogin/Screens/Logged in/PremiumPerksView.swift

```diff
@@ -18,43 +18,74 @@
 // You should have received a copy of the GNU General Public License
 // along with SimpleLogin. If not, see https://www.gnu.org/licenses/.
 
+import Combine
 import SwiftUI
 import SimpleLoginPackage
+import Shared
 
 struct PremiumPerksView: View {
-    var onUpgrade: () -> Void
+    @StateObject private var viewModel = PremiumPerksViewModel()
+
+    var onUpgrade: (Shared.Subscriptions) -> Void
+    var onRestore: () -> Void
 
     var body: some View {
         VStack(spacing: 0) {
             Text("Discover Premium")
                 .font(.title.bold())
                 .padding(.bottom)
-
             Spacer()
-
             perks
-
             Spacer()
 
-            Button(action: onUpgrade) {
-                Text("Upgrade")
-                    .frame(maxWidth: .infinity)
-                    .font(.headline)
-                    .padding(8)
+            Group {
+                upgradeButton
+                restoreButton
             }
-            .frame(maxWidth: .infinity)
-            .buttonStyle(.borderedProminent)
-            .tint(Color.blue)
-            .padding(.top)
+            .disabled(viewModel.isGettingSubscriptions || viewModel.isRestoring)
         }
         .padding()
         .roundedBorderedBackground()
+        .onReceive(Just(viewModel.restorePurchaseState)) { state in
```

Comment:
Why not use the viewModel.$restorePurchaseState instead of wrapping viewModel.restorePurchaseState inside a Just publisher ?

Reply:
✅